pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -63,21 +87,58 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 	if !current.CanTransitionTo(next) {
 		return fmt.Errorf("invalid state transition: %s -> %s", current.Name(), next.Name())
 	}
-	// TODO: call pre-transition listeners -  Issue: https://github.com/hyperledger/aries-framework-go/issues/140
-	followup, err := next.Execute(msg)
+
+	// trigger the event if the message type is registered (if it's not outbound)
+	eventsTriggered := false
+	if !msg.Outbound {
+		eventsTriggered, err = s.sendEvent(msg, thid, next)
+		if err != nil {
+			return fmt.Errorf("send events failed: %w", err)
+		}
+	}
+
+	// if no events are trigger continue the execution
+	if !eventsTriggered {
+		return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
+	}
+
+	return nil
+}
+
+// Register events on DID Exchange protocol.
+func (s *Service) Register(msgType string, ch chan<- event.DIDCommEvent) error {
```

Comment:
Also what is hidden behind the pattern of passing a channel as an argument is that:
- the caller can control the channel buffer size.
- the caller can use the same channel against multiple protocols.